okhttp/src/main/java/okhttp3/internal/ws/WebSocketReader.kt

```diff
@@ -13,223 +13,225 @@
+    source.timeout().clearTimeout()
     try {
-      b0 = source.readByte() & 0xff;
+      b0 = source.readByte() and 0xff
     } finally {
-      source.timeout().timeout(timeoutBefore, TimeUnit.NANOSECONDS);
+      source.timeout().timeout(timeoutBefore, TimeUnit.NANOSECONDS)
     }
 
-    opcode = b0 & B0_MASK_OPCODE;
-    isFinalFrame = (b0 & B0_FLAG_FIN) != 0;
-    isControlFrame = (b0 & OPCODE_FLAG_CONTROL) != 0;
+    opcode = b0 and B0_MASK_OPCODE
+    isFinalFrame = b0 and B0_FLAG_FIN != 0
+    isControlFrame = b0 and OPCODE_FLAG_CONTROL != 0
 
     // Control frames must be final frames (cannot contain continuations).
     if (isControlFrame && !isFinalFrame) {
-      throw new ProtocolException("Control frames must be final.");
+      throw ProtocolException("Control frames must be final.")
     }
 
-    boolean reservedFlag1 = (b0 & B0_FLAG_RSV1) != 0;
-    boolean reservedFlag2 = (b0 & B0_FLAG_RSV2) != 0;
-    boolean reservedFlag3 = (b0 & B0_FLAG_RSV3) != 0;
+    val reservedFlag1 = b0 and B0_FLAG_RSV1 != 0
+    val reservedFlag2 = b0 and B0_FLAG_RSV2 != 0
+    val reservedFlag3 = b0 and B0_FLAG_RSV3 != 0
     if (reservedFlag1 || reservedFlag2 || reservedFlag3) {
       // Reserved flags are for extensions which we currently do not support.
-      throw new ProtocolException("Reserved flags are unsupported.");
+      throw ProtocolException("Reserved flags are unsupported.")
     }
 
-    int b1 = source.readByte() & 0xff;
+    val b1 = source.readByte() and 0xff
 
-    boolean isMasked = (b1 & B1_FLAG_MASK) != 0;
+    val isMasked = b1 and B1_FLAG_MASK != 0
     if (isMasked == isClient) {
       // Masked payloads must be read on the server. Unmasked payloads must be read on the client.
-      throw new ProtocolException(isClient
-          ? "Server-sent frames must not be masked."
-          : "Client-sent frames must be masked.");
+      throw ProtocolException(if (isClient) {
+        "Server-sent frames must not be masked."
+      } else {
+        "Client-sent frames must be masked."
+      })
     }
 
     // Get frame length, optionally reading from follow-up bytes if indicated by special values.
-    frameLength = b1 & B1_MASK_LENGTH;
-    if (frameLength == PAYLOAD_SHORT) {
-      frameLength = source.readShort() & 0xffffL; // Value is unsigned.
-    } else if (frameLength == PAYLOAD_LONG) {
-      frameLength = source.readLong();
+    frameLength = (b1 and B1_MASK_LENGTH).toLong()
+    if (frameLength == PAYLOAD_SHORT.toLong()) {
+      frameLength = (source.readShort() and 0xffff).toLong() // Value is unsigned.
+    } else if (frameLength == PAYLOAD_LONG.toLong()) {
+      frameLength = source.readLong()
       if (frameLength < 0) {
-        throw new ProtocolException(
-            "Frame length 0x" + Long.toHexString(frameLength) + " > 0x7FFFFFFFFFFFFFFF");
+        throw ProtocolException(
+            "Frame length 0x${java.lang.Long.toHexString(frameLength)} > 0x7FFFFFFFFFFFFFFF")
```

Comment:
Weird that there isn't a nice form of this that handles unsigned nums.  Guess that's coming.

Reply:
the unsigned bit is the gotcha. It’s tempting to do Long.toString(16) but it doesn’t do what we want.

Reply:
(I guess `toUlong().toString(16)` will be semantic)